extensions/oplogPopulator/allocationStrategy/RetainBucketsDecorator.js

```diff
@@ -0,0 +1,86 @@
+const AllocationStrategy = require('./AllocationStrategy');
+
+/**
+ * @class RetainBucketsDecorator
+ *
+ * @classdesc RetainBucketsDecorator is a decorator
+ * that retains buckets that are removed from the
+ * connector but still in use.
+ */
+class RetainBucketsDecorator extends AllocationStrategy {
+    /**
+     * @constructor
+     * @param {AllocationStrategy} strategy the strategy to decorate
+     * @param {Object} params params
+     * @param {Logger} params.logger logger object
+     */
+    constructor(strategy, params) {
+        super(params);
+        this._strategy = strategy;
+
+        // Stores buckets that should are removed from the connector
+        // but still in use
+        this._retainedBuckets = new Map();
+    }
+
+    /**
+     * Callback when a bucket is removed from a connector
+     * @param {String} bucket bucket name
+     * @param {Connector} connector connector
+     * @returns {undefined}
+     */
+    onBucketRemoved(bucket, connector) {
+        // When a bucket is removed, it is not immediately removed
+        // from the connector. This allows to retain the bucket
+        // in the connector until it is removed.
+        this._retainedBuckets.set(bucket, connector);
+    }
+
+    /**
+     * Callback when a connector is updated
+     * @param {Connector} connector connector
+     * @returns {undefined}
+     */
+    onConnectorUpdated(connector) {
+        // When a connector is updated or destroyed, the retained
```

Comment:
there is a single code, but the way it is integrated can cause confusion...
may be more readable to have the "public" calkback (onConnectorDestroy), for binding in oplog populator ; and an internal function to do the actual removal on destroy or update

```suggestion
    onConnectorDestroy(connector) {
        this._cleanupRetainedBuclet(connector)
    }

    _cleanupRetainedBuclet(connector) {
        // When a connector is updated or destroyed, the retained
```